### PR TITLE
sets basic auth

### PR DESCRIPTION
**Pull request by ivancortesromero**
- According to Ruby doc, [use_ssl](https://ruby-doc.org/stdlib-2.5.1/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-HTTPS) enables HTTPS

- Adds Basic Authorization

- Upgrade to ruby 2.5.1

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export SOURCE_USER='xxxxx'
+export SOURCE_PASSWORD='xxxxxxxxx'
```
Comment (do-diegoortiz) [inline .env.sample]: Para que sirve ese `export` ? 🤔

File: Gemfile.lock [hunk 2]
```diff
@@ -212,7 +212,7 @@ DEPENDENCIES
   web-console (>= 3.3.0)
 
 RUBY VERSION
-   ruby 2.3.1p112
+   ruby 2.5.1p57
```
Comment (do-diegoortiz) [inline Gemfile.lock]: Qué significa esa p57?

File: app/models/machine.rb [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
+
+    @doc = REXML::Document.new(res.body)
+
+    response
+  end
+
+  private
+
+  def self.response
+    {}.tap do |r|
```
Comment (do-diegoortiz) [inline app/models/machine.rb]: Uichhhh no había visto esos { } vacios. Interesante.

File: Gemfile.lock (hunk 2 above)
Comment (ivancortesromero) [reply Gemfile.lock]: It means a `patch`, a sublevel into the (semantic)[https://semver.org] of versions.

**Pull request merged**